storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -96,56 +111,132 @@ func (s *commitLogSource) Read(
 	defer iter.Close()
 
 	var (
-		namespace   = ns.ID()
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
-		bopts       = s.opts.ResultOptions()
-		blopts      = bopts.DatabaseBlockOptions()
-		blockSize   = ns.Options().RetentionOptions().BlockSize()
-		encoderPool = bopts.DatabaseBlockOptions().EncoderPool()
-		errs        = 0
+		namespace    = ns.ID()
+		numShards    = len(shardsTimeRanges)
+		unmerged     = []map[ts.Hash]encodersByTime{}
+		unmergedLock = &sync.RWMutex{}
+		numConc      = s.opts.M3TSZEncodingConcurrency()
+		bopts        = s.opts.ResultOptions()
+		blopts       = bopts.DatabaseBlockOptions()
+		blockSize    = ns.Options().RetentionOptions().BlockSize()
+		encoderPool  = bopts.DatabaseBlockOptions().EncoderPool()
+		workerErrs   = make([]int, numConc)
 	)
+
+	encoderChans := []chan encoderArg{}
```

Comment:
nit: Create with upfront size to save allocation here, i.e. `make([]chan encoderArg, 0, numConc)`